pkg/volume/csi/csi_client.go

```diff
@@ -31,12 +31,42 @@ import (
 	"k8s.io/kubernetes/pkg/features"
 )
 
-type csiClient interface {
+type csiNodeInfoGetter interface {
 	NodeGetInfo(ctx context.Context) (
 		nodeID string,
 		maxVolumePerNode int64,
 		accessibleTopology *csipb.Topology,
 		err error)
+	// for pre-v1.0
+	NodeGetId(ctx context.Context) (nodeID string, err error)
+}
+
+// getNodeInfo tries to call NodeGetInfo() and falls back to NodeGetId() when
+// the former is not implemented.
+//
+// NodeGetId() is only called when NodeGetInfo() returns an "unknown method"
+// error, all other errors are immediately bubbled up and NodeGetId() is not
+// used at all.
+//
+// TODO(hoegaarden): e2e test with a driver which only supports NodeGetId() but
+//                   not NodeGetInfo()
+func getNodeInfoWithFallback(ctx context.Context, c csiNodeInfoGetter) (string, int64, *csipb.Topology, error) {
```

Comment:
nit: either name return values or return a named struct. Without that these types are hard to figure out